tests/Tokenizer/Analyzer/AttributeAnalyzerTest.php

```diff
@@ -129,4 +130,286 @@ public static function provideIsAttributeCases(): iterable
 
         yield [true, '<?php #[Foo("(")] class Bar {}'];
     }
+
+    /**
+     * @requires PHP 8.0
+     *
+     * @dataProvider provideGetAttributeDeclarationsCases
+     *
+     * @param array{0: string, 1: list<AttributeAnalysis>} $expected
+     */
+    public function testGetAttributeDeclarations(string $code, int $startIndex, array $expected): void
+    {
+        $tokens = Tokens::fromCode($code);
+        $attributeDeclarationAnalyses = AttributeAnalyzer::collect($tokens, $startIndex);
+
+        self::assertSame(
+            serialize($expected),
+            serialize($attributeDeclarationAnalyses),
+        );
+    }
+
+    /**
+     * @return iterable<array{0: string, 1: int, 2: list<AttributeAnalysis>}>
+     */
+    public static function provideGetAttributeDeclarationsCases(): iterable
+    {
+        yield [
```

Comment:
Can we have names for the cases? Took me a "minute" to notice some of them are multiple attributes, and some are attributes with multiple arguments.

Reply:
```suggestion
        yield 'multiple #[] in a multiline group' => [
```

Reply:
Done